jwst/pipeline/calwebb_coron3.py

```diff
@@ -118,50 +130,16 @@ def process(self, input):
                 suffix='psfsub', acid=acid
             )
 
-            # Create a ModelContainer of the psf_sub results to send to
-            # outlier_detection
-            self.log.debug('Building ModelContainer of klip results')
-            target_models = datamodels.ModelContainer()
-            for i in range(psf_sub.data.shape[0]):
-                image = datamodels.ImageModel(data=psf_sub.data[i],
-                                              err=psf_sub.err[i],
-                                              dq=psf_sub.dq[i])
-                image.update(psf_sub)
-                image.meta.wcs = psf_sub.meta.wcs
-                target_models.append(image)
-
-            # Call outlier_detection
-            target_models = self.outlier_detection(target_models)
-
-            # Create Level 2c products
-            if target_models[0].meta.cal_step.outlier_detection == 'COMPLETE':
-                err_str1 = "Creating Level 2c output with updated DQ arrays..."
-                self.log.info(err_str1)
-                lev2c_model = psf_sub.copy()
-                # Replace Level 2b product DQ array with Level 2c DQ array
-                for i in range(len(target_models)):
-                    lev2c_model.dq[i] = target_models[i].dq
-                lev2c_model.meta.cal_step.outlier_detection = 'COMPLETE'
-                self.save_model(lev2c_model, output_file=target_file,
-                                suffix='crfints', acid=acid)
-
-            # Append results from this target exposure to resample input model
-            for i in range(len(target_models)):
-                resample_input.append(target_models[i])
+            # Split out the integrations into separate models
+            # in a ModelContainer to pass to `resample`
+            for model in psf_sub.to_container():
+                resample_input.append(model)
```

Comment:
Can you just do

```
result = self.resample(psf_sub.to_container())
```
instead?

Reply:
The resample step is outside of the loop and is called only once to operate on all input science files. This is to pass to resample all the integrations of all the input sciences as a flat list.

Reply:
Ah, I missed the indentation. 👍